test/PrizeSplit.test.js

```diff
@@ -0,0 +1,507 @@
+
+  let prizePeriodStart = now();
+  let prizePeriodSeconds = 1000;
+
+  beforeEach(async () => {
+    [
+      wallet,
+      wallet2,
+      wallet3,
+      wallet4,
+      wallet5,
+      wallet6,
+    ] = await hardhat.ethers.getSigners();
+
+    debug({
+      wallet: wallet.address,
+      wallet2: wallet2.address,
+      wallet3: wallet3.address,
+      wallet4: wallet4.address,
+    });
+
+    debug("mocking tokens...");
+    const Ticket = await hre.artifacts.readArtifact("ControlledToken");
+    ticket = await deployMockContract(wallet, Ticket.abi, overrides);
+
+    const ControlledToken = await hre.artifacts.readArtifact("ControlledToken");
+    sponsorship = await deployMockContract(
+      wallet,
+      ControlledToken.abi,
+      overrides
+    );
+
+    debug("deploying prizePool...");
+    const PrizePool = await hre.artifacts.readArtifact("PrizePool");
+    prizePool = await deployMockContract(wallet, PrizePool.abi, overrides);
+
+    debug("deploying prizeSplit...");
+    const PrizeSplitHarness = await hre.ethers.getContractFactory(
+      "PrizeSplitHarness",
+      wallet,
+      overrides
+    );
+
+    prizeSplitHarness = await PrizeSplitHarness.deploy();
+
+    debug("initialized!");
+  });
+
+
+  describe("setPrizeSplits()", () => {
+    it("should revert with invalid prize split target address", async () => {
+      await expect(
+        prizeSplitHarness.setPrizeSplits([
+          {
+            target: constants.AddressZero,
+            percentage: "100",
+            token: 0,
+          },
+        ])
+      ).to.be.revertedWith(
+        "MultipleWinners/invalid-prizesplit-target"
+      );
+    });
+
+    it("should revert with single prize split config is equal to or above 100% percent", async () => {
+
+      await expect(
+        prizeSplitHarness.setPrizeSplits([
+          {
+            target: wallet5.address,
+            percentage: "1005",
+            token: 0,
+          },
+        ])
+      ).to.be.revertedWith(
+        "MultipleWinners/invalid-prizesplit-percentage"
+      );
+    });
+
+    it("should revert when multuple prize split configs is above 100% percent", async () => {
+      await expect(
+        prizeSplitHarness.setPrizeSplits([
+          {
+            target: wallet5.address,
+            percentage: 500,
+            token: 0,
+          },
+          {
+            target: wallet6.address,
+            percentage: 501,
+            token: 0,
+          },
+        ])
+      ).to.be.revertedWith("MultipleWinners/invalid-prizesplit-percentage-total");
+    });
+
+    it("should revert with invalid prize split token enum", async () => {
+      await expect(
+        prizeSplitHarness.setPrizeSplits([
+          {
+            target: wallet5.address,
+            percentage: 500,
+            token: 2,
+          },
+          {
+            target: constants.AddressZero,
+            percentage: "0",
+            token: 0,
+          },
+        ])
+      ).to.be.reverted
```

Comment:
It should be reverted with an error message right?

Reply:
Since it was enforced by an enum type it actually didn't revert with a message but an invalid opcode instead.